addons/binding/org.openhab.binding.mcp23017/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="mcp23017" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="mcp23017switch">
+		<label>mcp23017 switch</label>
+		<description>Thing for mcp23017 pin working as a switch (mode DIGITAL_OUTPUT)</description>
+
+		<channels>
+			<channel id="switch" typeId="switch" />
+		</channels>
+
+		<config-description>
+			<parameter name="address" type="integer" required="true">
+				<label>I2C Device Address</label>
+				<description>I2C Bus mcp2317 device Address</description>
+				<default>20</default>
+			</parameter>
+            <parameter name="bus_number" type="integer" required="true">
```

Comment:
Formatting looks inconsistent, I suppose it is a combination of tabs and spaces, please reformat with the ESH profile in Eclipse.